File: Content/Java/utils/Reader.java
import java.io.*;
import java.math.*;
import java.util.*;

public class Reader {
  private BufferedReader in; private StringTokenizer st;
  public Reader(InputStream inputStream) {
    in = new BufferedReader(new InputStreamReader(inputStream));
  }
  public Reader(String fileName) throws FileNotFoundException {
    in = new BufferedReader(new FileReader(fileName));
  }
  public String next() throws IOException {
    while (st == null || !st.hasMoreTokens())
      st = new StringTokenizer(in.readLine().trim());
    return st.nextToken();
  }
  public String nextLine() throws IOException {
    st = null; return in.readLine();
  }
  public BigInteger nextBigInteger() throws IOException {
    return new BigInteger(next());
  }
  public byte nextByte() throws IOException {
    return Byte.parseByte(next());
  }
  public char nextChar() throws IOException { return next().charAt(0); }
  public double nextDouble() throws IOException {
    return Double.parseDouble(next());
  }
  public float nextFloat() throws IOException {
    return Float.parseFloat(next());
  }
  public int nextInt() throws IOException {
    return Integer.parseInt(next());
  }
  public long nextLong() throws IOException {
    return Long.parseLong(next());
  }
  public short nextShort() throws IOException {
    return Short.parseShort(next());
  }
  public void close() throws IOException { in.close(); }
}

File: Content/Kotlin/utils/Reader.kt
import java.io.*
import java.math.*
import java.util.*

class Reader {
  private val In: BufferedReader
  private var st: StringTokenizer? = null
  constructor(inputStream: InputStream) {
    In = BufferedReader(InputStreamReader(inputStream))
  }
  constructor(fileName: String) {
    In = BufferedReader(FileReader(fileName))
  }
  fun next(): String {
    while (st == null || !st!!.hasMoreTokens())
      st = StringTokenizer(In.readLine().trim())
    return st!!.nextToken()
  }
  fun nextLine(): String {
    st = null
    return In.readLine()
  }
  fun nextChar(): Char = next()[0]
  fun nextDouble(): Double = next().toDouble()
  fun nextInt(): Int = next().toInt()
  fun nextLong(): Long = next().toLong()
  fun close(): Unit = In.close()
}
